src/Backend/Modules/Pages/Domain/Page/Form/PageDataRedirectType.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Backend\Modules\Pages\Domain\Page\Form;
+
+use Backend\Modules\Pages\Engine\Model as BackendPagesModel;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\UrlType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class PageDataRedirectType extends AbstractType implements PageDataTypeInterface
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder->add(
+            'redirect',
+            ChoiceType::class,
+            [
+                'label' => 'lbl.Redirect',
+                'choices' => [
+                    'lbl.None' => 'none',
+                    'lbl.InternalLink' => 'internal',
+                    'lbl.ExternalLink' => 'external',
+                ],
+                'label_attr' => ['class' => 'custom-control-label radio-custom'],
+                'expanded' => true,
+                'attr' => [
+                    'class' => 'radiobuttonFieldCombo'
+                ],
+            ]
+        );
+        $builder->add(
+            'internal_redirect',
+            ChoiceType::class,
+            [
+                'choices' => array_flip(BackendPagesModel::getPagesForDropdown()),
+                'label' => 'lbl.InternalLink',
+                'label_attr' => [
+                    'class' => 'sr-only',
+                ],
+            ]
+        );
+        $builder->add(
+            'external_redirect',
+            UrlType::class,
+            [
+                'label' => 'lbl.ExternalLink',
+                'label_attr' => [
+                    'class' => 'sr-only',
+                ],
+            ]
+        );
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults(
+            [
+                'inherit_data' => true,
+            ]
+        );
+    }
+
+    public static function transform(?array $persistedData, array $transformedData): array
+    {
+        $transformedData['redirect'] = 'none';
+        $transformedData['internal_redirect'] = null;
+        $transformedData['external_redirect'] = null;
+
+        if ($persistedData === null) {
+            return $transformedData;
+        }
+
+        if (isset($persistedData['internal_redirect']['page_id'])) {
+            $transformedData['redirect'] = 'internal';
+            $transformedData['external_redirect'] = $persistedData['internal_redirect']['page_id'];
+        } elseif (isset($persistedData['external_redirect']['url'])) {
```

Comment:
Please use early returns here like you did above with the persisted data

Reply:
fixed in f22bc35e3